xen/arch/arm/coproc/coproc.h

```diff
@@ -136,6 +136,44 @@ struct vcoproc_instance {
     void *priv;
 };
 
+/* coproc logger functions */
+
+extern int coproc_debug;
+
+#define coproc_dev_print(dev, lvl, coproc_lvl, fmt, ...)                       \
+do                                                                             \
+{                                                                              \
+    if ( coproc_lvl <= coproc_debug )                                          \
+        printk(lvl "coproc: %s: " fmt,                                         \
+               dev ? dt_node_full_name(dev_to_dt(dev)) : "",                   \
+               ## __VA_ARGS__);                                                \
+} while (0)
+
+/*
+ * this is defined for convenience so we don't have to translate
+ * XENLOG_XXX into corresponding numbers at runtime
+ */
+enum COPROC_DBG_LEVEL
+{
+    COPROC_DBG_ERROR,
+    COPROC_DBG_WARN,
+    COPROC_DBG_INFO,
+    COPROC_DBG_DEBUG,
+    COPROC_DBG_VERB,
+    COPROC_DBG_LAST
+};
+
+#define COPROC_ERROR(dev, fmt, ...)                                            \
```

Comment:
as you explained orally you like to keep caps as eye catcher
SCF_XXX looks good to me